charts/redpanda/templates/statefulset.yaml

```
@@ -331,7 +322,7 @@ spec:
 {{- if gt ( .Values.statefulset.replicas | int64 ) 2 }}
         - name: lifecycle-scripts
           secret:
-            secretName: {{ template "redpanda.fullname" . }}-sts-lifecycle
+            secretName: {{ trunc 50 (include "redpanda.fullname" .) }}-sts-lifecycle
```

NIT: Just to be consistent
```suggestion
            secretName: {{ (include "redpanda.fullname" . | trunc 50 ) }}-sts-lifecycle
```